File: SoCPhuong.py
'''
Nhập vào 1 số.

Kiểm tra số đó có phải là số chính phương không?
Xuất: yes, nếu đó là số chính phương, ngược lại xuất no

ví dụ:
input: 4

output: yes
'''

n = int(input())
def scp(n):
    if n < 0:
        return False
    x = int(n ** 0.5)
    return x * x == n
if(scp(n)):
    print("yes")
else:
    print("no")
    
# Cách 2:
n = int(input())
if n < 0:
    print("no")
else:
    x = int(n ** 0.5)
    if x * x == n:
        print("yes")
    else:
        print("no")

File: LTCB/C++/Ham/Revise/SoCPhuong.cpp
/*
Nhập vào 1 số.

Kiểm tra số đó có phải là số chính phương không?
Xuất: yes, nếu đó là số chính phương, ngược lại xuất no

ví dụ:
input: 4

output: yes
*/

#include <iostream>
#include <cmath>
using namespace std;

int scp(int n){
    int x = sqrt(n);
    if(x*x == n) return 1;
    return 0;
}

int main(){
    int n;
    cin >> n;
    if(scp(n)) cout << "yes";
    else cout << "no";
    return 0;
}